lib/tunnel.js

```diff
@@ -0,0 +1,59 @@
+var net = require('net');
+var ssh2 = require('ssh2');
+
+module.exports = makeTunnel;
+
+function makeTunnel(urlObj, callback) {
+  var conn = new ssh2.Client();
+  var env = process.env;
+  var username = env.LOGNAME || env.USER || env.LNAME || env.USERNAME;
+
+  conn.on('ready', function() {
+    makeProxy(conn, urlObj, callback);
+    // don't keep the process alive if all we have is this connection
+    conn._sock.unref();
+  }).on('error', function(err) {
+    console.error('ssh connection error: ', err);
+  }).connect({
+    // extract hostname from url
+    host: urlObj.hostname,
+    // assume ssh is on port 22, not configurable (yet?)
+    port: 22,
+    // assume ssh user is the same as current user
+    username: username,
+    // assume an ssh agent is already running
+    agent: process.env.SSH_AUTH_SOCK,
```

Comment:
I'm hoping this is not actually necessary, its going to degrade the "if you can ssh without a password, you can use our ssh tunnelling" story.